pkg/virt-operator/resource/generate/components/deployments.go

```diff
@@ -303,12 +303,14 @@ func newPodAntiAffinity(key, topologyKey string, operator metav1.LabelSelectorOp
 	}
 }
 
-func NewApiServerDeployment(namespace string, repository string, imagePrefix string, version string, productName string, productVersion string, productComponent string, pullPolicy corev1.PullPolicy, verbosity string, extraEnv map[string]string) (*appsv1.Deployment, error) {
+func NewApiServerDeployment(namespace, repository, imagePrefix, version, productName, productVersion, productComponent, image string, pullPolicy corev1.PullPolicy, verbosity string, extraEnv map[string]string) (*appsv1.Deployment, error) {
 	podAntiAffinity := newPodAntiAffinity(kubevirtLabelKey, kubernetesHostnameTopologyKey, metav1.LabelSelectorOpIn, []string{VirtAPIName})
 	deploymentName := VirtAPIName
-	imageName := fmt.Sprintf("%s%s", imagePrefix, deploymentName)
+	if image == "" {
+		image = fmt.Sprintf("%s%s", imagePrefix, deploymentName)
+	}
```

Comment:
I've heard people claiming that before but never really understood why :) can you the motivation for not overriding parameters?

Anyways, this is now removed for other reasons :)

Reply:
There isn't a strong motivation behind this, basically, it's only a matter of style. I usually prefer to leave the input parameter as is. I think that it makes debugging and code reading easier to understand, at least for me. My cognitive process puts the inputs to a function in a closed box, and overwriting a value causes me to have to remember two things instead of one; which makes it seem like it's harder than having two separate variables. :) Clearly this is not a big deal, and I would never block a PR for that reason :)
This is the most motivation I can find :D

Reply:
Well, fair enough :)
The way I look at it is that by-value parameters (unlike pointers) are variables that you can mutate as much as you'd like without being worried by side-effects. But I guess different styles are clearer to different people :)